### PR TITLE
Admin|| register links to entity TC-1

### PR DESCRIPTION
**Pull request by Marina-sukhova**
https://trello.com/c/lN4aO5mp/954-admin-link-register-to-entity-tc

### CODE REVIEW

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 1]
```diff
@@ -110,12 +122,34 @@ public ConfigurationPage clickEntities() {
...
+    public AdminFieldsPage clickRegister(String specification) {
+        registers.click();
+        getDriver().findElement(By.xpath(String.format(anyRegister, specification))).click();
+        return new AdminFieldsPage(getDriver());
+    }
+    public AdminCreateRegisterPage clickNewRegister(String specification) {
+        registers.click();
+        getDriver().findElement(By.xpath(String.format(anyRegister, specification))).click();
+        return new AdminCreateRegisterPage(getDriver());
+    }
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:134]: I do not see any difference between these methods except return statement, so `clickNewRegister` should click on something else

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 2]
```diff
@@ -101,6 +102,17 @@
     @FindBy(xpath = "//span[contains(text(), 'UI')]")
     private WebElement permissionsUI;
 
+    @FindBy(xpath = "//p[contains(text(), 'Registers')]")
+    private WebElement registers;
+
+    private String anyRegister = "//div[@id='formsRegisters']//span[contains(text(), '%s')]";
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:108]: create it as `private static final` and replace it to the top of the class

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 3]
```diff
@@ -101,6 +102,17 @@
...
+
+
+    @FindBy(xpath = "//ul[@class = 'navbar-nav']//i[contains(., 'app_registration')]")
+    private WebElement registerIcon;
+
+    private String createdEntity = "//span[contains(., '%s')]";
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:114]: This is a duplicate of existed `protected static final String configEntityName = "//span[contains(text(),'%s')]";`, hence delete it

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 4]
```diff
@@ -110,12 +122,34 @@ public ConfigurationPage clickEntities() {
...
         clickEntities();
         newEntity.click();
         return new AdminCreateEntityPage(getDriver());
     }
 
+    public ConfigurationPage clickCreatedEntity(String EntityName) {
+        clickEntities();
+        getDriver().findElement(By.xpath(String.format(createdEntity, EntityName))).click();
+        return new AdminCreateEntityPage(getDriver());
+    }
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:146]: This is a duplicate of existed `clickConfigEntity(String entityName)`, hence delete it

File: src/test/java/model/admin/configuration/entities/AdminCreateEntityPage.java [hunk 5]
```diff
@@ -1,6 +1,7 @@
 package model.admin.configuration.entities;
 
 import model.admin.common.ConfigurationPage;
+import model.admin.configuration.registers.AdminCreateRegisterPage;
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminCreateEntityPage.java:4]: Is it really needed here?

File: src/test/java/model/admin/configuration/entities/AdminFieldsPage.java [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 import com.beust.jcommander.Strings;
 import model.admin.common.ConfigurationPage;
+import model.admin.configuration.registers.AdminCreateRegisterPage;
```
Comment (oooleks) [inline src/test/java/model/admin/configuration/entities/AdminFieldsPage.java:5]: It looks like there is no need of that import here

File: src/test/java/model/admin/edit/EntityEditPage.java [hunk 7]
```diff
@@ -46,7 +48,13 @@
     @FindBy(xpath = "//input[@id = 'datetime']")
     private WebElement inputDateTime;
 
-    @Override
+    @FindBy(id = "article")
+    private WebElement article;
+
+    @FindBy(id = "quantity")
+    private WebElement quantity;
```
Comment (oooleks) [inline src/test/java/model/admin/edit/EntityEditPage.java:55]: Do not look for elements using test data from the test ("article", "quantity")

File: src/test/java/test/admin/AdminRegisterTest.java [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+                .mapFieldType("Negative (-)")
+                .mapFieldQuantity(entityFieldsLable[1])
+                .mapFieldCondition("Always enabled")
+                .mapFieldName(entityFieldsLable[0]);
+
+                Assert.assertTrue(new ConfigurationPage(getDriver())
+                        .clickConfiguration()
+                        .clickCreatedEntity(positiveEntityName)
+                        .clickRegisterIcon().getTableValue(registerName));
+
```
Comment (oooleks) [inline src/test/java/test/admin/AdminRegisterTest.java:101]: Align the code and delete extra lines when finished

**Review by guram21 — APPROVED** [review]

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 4 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java:146]: That's a good point. Fixed that

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 3 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java:114]: fixed that

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 2 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java:108]: done. thanks

File: src/test/java/model/admin/configuration/entities/AdminCreateEntityPage.java (hunk 5 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/configuration/entities/AdminCreateEntityPage.java:4]: test doesn't work without this import

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 9]
```diff
@@ -25,6 +26,8 @@
 
     protected static final String configEntityName = "//span[contains(text(),'%s')]";
     protected static final By cmdLineElement = By.xpath("//textarea[@id='pa-cli-cmd']");
+    private final String createdRegister = "//div[@id='formsRegisters']//span[contains(text(), '%s')]";
+    private final String createdEntity = "//span[contains(., '%s')]";
```
Comment (oooleks) [inline src/test/java/model/admin/common/ConfigurationPage.java:30]: your #30 is = to existed #27

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 1 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java]: got you! thanks

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 9 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java:29]: fixed that. thanks

File: src/test/java/model/admin/edit/EntityEditPage.java (hunk 7 above)
Comment (Marina-sukhova) [reply src/test/java/model/admin/edit/EntityEditPage.java]: got it

File: src/test/java/model/admin/common/ConfigurationPage.java [hunk 10]
```diff
@@ -101,6 +104,15 @@
     @FindBy(xpath = "//span[contains(text(), 'UI')]")
     private WebElement permissionsUI;
 
+    @FindBy(xpath = "//p[contains(text(), 'Registers')]")
+    private WebElement registers;
+
+    @FindBy(xpath =  "//div[@id='formsRegisters']//span[contains(text(), 'New')]")
+    private WebElement NewRegister;
```
[the remark this answers is not in the document]
Comment (Marina-sukhova) [reply src/test/java/model/admin/common/ConfigurationPage.java:117]: fixed that.

File: src/test/java/model/admin/common/ConfigurationPage.java (hunk 9 above)
Comment (oooleks) [reply src/test/java/model/admin/common/ConfigurationPage.java:30]: No change so far. Will you change it in this PR or defer it?

**Review by oooleks — APPROVED** [review]

**Pull request merged**